server/src/subscriber/subscriber.controller.ts

```diff
@@ -121,4 +121,26 @@ export class SubscriberController {
 
     response.send(updatedContact);
   }
+
+  @Get("/subscribers/email/:email")
+  async getEmail(@Param() params, @Res() response) {
+    if (!validateEmail(params.email)) {
+      response.status(400).send({
+        error: "Invalid email address."
+      })
+      return;
+    }
+
+    const existingUser = await this.subscriberService.findByEmail(params.email);
+    const existingUserListId = `zap_${this.sendgridEnvironment}_id`;
+    if (existingUser.code === 404 || existingUser['1'].result[params.email].contact.custom_fields[existingUserListId]) {
```

Comment:
Let's have this also check that the user is associated with the Sendgrid List.

Reply:
Forgot to negate the second condition